### PR TITLE
Docs: Improve documentation for the public api

### PR DESCRIPTION
**Pull request by DerMolly**
### Component/Part
apidocs

### Description
This PR improves documentation for the public api

### Steps

- [x] Added / updated documentation
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

**Comment by DerMolly** [comment]
As far as I can tell from running https://github.com/OpenAPITools/openapi-diff with the generated api spec (`http://localhost:3000/apidoc-json`) and the handcrafted one the differences are still:
- `GET /api/v2/notes/{note}/media` is missing
- the Dtos changed on multiple occasions.
- `POST /api/v2/media` does not communicate the different types of images that are supported.

File: src/api/public/me/me.controller.ts [hunk 1]
```diff
@@ -65,6 +87,12 @@ export class MeController {
 
   @UseGuards(TokenAuthGuard)
   @Get('history/:note')
+  @ApiOkResponse({
+    description: 'The history entry of the user and note',
```
Comment (InnayTool) [inline src/api/public/me/me.controller.ts:91]:
```suggestion
    description: 'The history entry of the user which points to the note',
```

File: src/api/public/notes/notes.controller.ts [hunk 2]
```diff
@@ -284,6 +359,13 @@ export class NotesController {
 
   @UseGuards(TokenAuthGuard)
   @Get(':noteIdOrAlias/revisions/:revisionId')
+  @ApiOkResponse({
+    description: 'Revision of the note for the given id.',
```
Comment (InnayTool) [inline src/api/public/notes/notes.controller.ts:363]:
```suggestion
    description: 'Revision of the note for the given id or alias.',
```

**Review by InnayTool — CHANGES_REQUESTED** [review]
Should the descriptions of the routes have a `.` at the end or not
(like
```
@ApiHeader({
    name: 'HedgeDoc-Note',
    description: 'ID or alias of the parent note',
})
```
or
```
@ApiHeader({
    name: 'HedgeDoc-Note',
    description: 'ID or alias of the parent note.',
})
```
)?
This seems to be inconsistent.

**Comment by DerMolly** [comment]
> Should the descriptions of the routes have a `.` at the end or not
> 
> This seems to be inconsistent in the PR.

Well it's inconsistent in the base docs. I'd myself go with no periods.

**Review by InnayTool — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

**Pull request merged**